src/main/java/com/OrganizationController.java

```diff
@@ -60,10 +95,26 @@ protected void doDelete(HttpServletRequest request, HttpServletResponse response
     }
 
     @Override
-    protected void doPut(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    protected void doPut(HttpServletRequest request, HttpServletResponse response) throws ServletException,
+            IOException {
 
-        int id = Integer.parseInt(request.getParameter("id"));
-        String name = request.getParameter("name");
+        StringBuilder org = new StringBuilder();
+        BufferedReader reader = request.getReader();
+
+        try {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                org.append(line).append("\n");
+            }
+        } finally {
+            reader.close();
+        }
+
+        Gson gson = new Gson();
+        Organization updatedOrganization = gson.fromJson(org.toString(), Organization.class);
+
+        int id = updatedOrganization.getId();
+        String name = updatedOrganization.getName();
         Organization organization = organizationService.updateOrg(id,name);
```

Comment:
better of you could split the variable names I guess. 

`updatedOrganization ` -> `organization`
`organization` -> `updatedOrganization`

What do you think?